port/js/src/messgen.js

```diff
@@ -561,11 +561,11 @@ export function initializeMessages(messagesJson, headerSchema) {
         HEADER_STRUCT: headerSchema ? new Struct(headerSchema) : HEADER_STRUCT
     };
 
-    for (let m in messagesJson) {
+    for (let m in getKeysWithSortById(messagesJson)) {
 
         let name = m.trim(),
```

Comment:
Doesn't work:
![image](https://user-images.githubusercontent.com/5452251/229441842-2e95fa29-a1ae-46f1-bfb1-964c6d4c349f.png)
Could you add some tests for this please ?